### ISSUE TITLE
Custom losses (e.g. H^1-norm)

### ISSUE THREAD

**Issue by pescap** [lululxvi/deepxde#467]
Hi, 
I was trying to reproduce the method proposed in: [Gradient-enhanced physics-informed neural networks for forward and inverse PDE problems](https://arxiv.org/pdf/2111.02801.pdf), for example using H^1 (Sobolev) norm instead of a L^2 norm. 

Generally, the losses take (y_true, y_pred) as inputs. 

Is there a simple way to incorporate x to the losses in order to evaluate gradient based losses/norms?

In parallel, it could be interesting to enable "custom losses" for model.compile(), i.e. something like 
`model.compile(loss_fn = custom_loss)`
with a previously defined `custom_loss(y_true, y_pred)` or even `custom_loss(y_true, y_pred, x)`

Thank you for your help! 
Paul

**Comment by lululxvi** [comment]
The code of gPINN will be public soon.

DeepXDE already supports it:
```
def custom_loss(y_true, y_pred):
    ...

model.compile(loss=custom_loss)
```
But not custom_loss(y_true, y_pred, x).
